File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPLastDeathLocation.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPLastDeathLocation implements PlayerStateProperty<Location> {
    @Override
    public String getConfigName() {
        return "last_death_location";
    }

    @Override
    public Location getDefaultSerializedProperty() {
        return null;
    }

    @Override
    public Location getSerializedPropertyFromPlayer(Player player) {
        return player.getLastDeathLocation();
    }

    @Override
    public void restoreProperty(Player player, Location deathLocation) {
        player.setLastDeathLocation(deathLocation);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPLastDeathLocation.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.Location
import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPLastDeathLocation extends PlayerStateProperty[Location]:
  override def getConfigName: String = "last_death_location"

  override def getDefaultSerializedProperty: Location = null

  override def getSerializedPropertyFromPlayer(player: Player): Location =
    player.getLastDeathLocation

  override def restoreProperty(player: Player, deathLocation: Location): Unit =
    player.setLastDeathLocation(deathLocation)